t/01-compile-check-all.t

```diff
@@ -13,36 +13,16 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
-# require all modules to check if they even load
-
 use strict;
 use warnings;
 
-use Test::Compile;
-use Mojo::File 'tempdir';
-
-my $tempdir;
-BEGIN {
-    unshift @INC, 'lib';
-
-    # FIXME: Requiring OpenQA::Worker::Cache::Service
-    # Sets up Minion with Mojo::SQLite, that on DESTROY automatically disconnects from the database.
-    # If the database is not existant or can't be accessed we get a warning, that translates to test failure.
-
-    $tempdir = tempdir;
-    $ENV{OPENQA_CACHE_DIR} = $tempdir;
```

Comment:
Yes, it is just `all_perl_files_ok` with one explicitly excluded file. When the other PR has been merged you can observe the Travis log and see that the checks are performed for all relevant files.

It needs to be skipped because it is a test module and as such requires modules of os-autoinst and the test distribution. I thought this would be obvious.

Reply:
Ok, seems like you actually can not see that in the Travis log because it doesn't run `prove` with `-v` flag. But locally it looks like this:

```
prove -v t/01-compile-check-all.t
t/01-compile-check-all.t .. 
ok 1 - Syntax check lib/OpenQA.pm
ok 2 - use strict   lib/OpenQA.pm
ok 3 - use warnings lib/OpenQA.pm
ok 4 - Syntax check lib/DBIx/Class/Timestamps.pm
ok 5 - use strict   lib/DBIx/Class/Timestamps.pm
ok 6 - use warnings lib/DBIx/Class/Timestamps.pm
[...]
k 511 - Syntax check lib/perlcritic/Perl/Critic/Policy/HashKeyQuotes.pm
ok 512 - use strict   lib/perlcritic/Perl/Critic/Policy/HashKeyQuotes.pm
ok 513 - use warnings lib/perlcritic/Perl/Critic/Policy/HashKeyQuotes.pm
ok 514 - Syntax check lib/perlcritic/Perl/Critic/Policy/ConsistentQuoteLikeWords.pm
ok 515 - use strict   lib/perlcritic/Perl/Critic/Policy/ConsistentQuoteLikeWords.pm
ok 516 - use warnings lib/perlcritic/Perl/Critic/Policy/ConsistentQuoteLikeWords.pm
ok 517 - Syntax check script/initdb
ok 518 - use strict   script/initdb
ok 519 - use warnings script/initdb
ok 520 - Syntax check script/client
ok 521 - use strict   script/client
ok 522 - use warnings script/client
ok 523 - Syntax check script/clone_job.pl
ok 524 - use strict   script/clone_job.pl
ok 525 - use warnings script/clone_job.p
[...]
ok 568 - Syntax check t/25-cache.t
ok 569 - use strict   t/25-cache.t
ok 570 - use warnings t/25-cache.t
ok 571 - Syntax check t/13-joblocks.t
ok 572 - use strict   t/13-joblocks.t
ok 573 - use warnings t/13-joblocks.t
ok 574 - Syntax check t/35-script_clone_job.t
ok 575 - use strict   t/35-script_clone_job.t
ok 576 - use warnings t/35-script_clone_job.t
ok 577 - Syntax check t/28-logging.t
ok 578 - use strict   t/28-logging.t
ok 579 - use warnings t/28-logging.t
[...]
ok 748 - Syntax check t/lib/OpenQA/SeleniumTest.pm
ok 749 - use strict   t/lib/OpenQA/SeleniumTest.pm
ok 750 - use warnings t/lib/OpenQA/SeleniumTest.pm
[...]
```

So all types of files and locations are covered - more than we had before.